base_local_planner/include/base_local_planner/goal_functions.h

```diff
@@ -53,6 +53,15 @@
 
 namespace base_local_planner {
 
+  /**
+   * Compute the direction of a point from the normal passing through the goal
```

Comment:
S: can u explain what this does a bit better?  I don't get it
And also what's the returned direction? +1 / -1?

Reply:
> can u explain what this does a bit better? I don't get it

I don't get it that well either :joy:

> And also what's the returned direction? +1 / -1?

Yes. Basically, we draw a line perpendicular to the vector AB (A=pose at goal index - 3, B= goal pose) and compute if a point is before or after that line. This was implemented by @ayushgaud. Personally this isn't how I would implement this, but it's how it's done in Path Follower and it works fine, so I just copied it.